front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/utils.ts

```diff
@@ -20,22 +25,31 @@ const getPathStyle = (
     backgroundColor?: string;
   };
 } => {
-  if (hovered && 'pathId' in hovered.element && path.id === hovered?.element.pathId && !dragging) {
-    return { color: PATH_COLORS.HOVERED_PATH, level: 1 };
+  const trainId = isOccurrenceId(path.id) ? extractPacedTrainIdFromOccurrenceId(path.id) : path.id;
+  const item = timetableItemsWithDetails?.find((t) => t.id === trainId);
+  const category = item?.category;
+
+  const colors = category ? TRAIN_CATEGORY_PATH_COLORS[category] : DEFAULT_TRAIN_PATH_COLORS;
+
+  const categoryColor = colors.normal;
+  const categoryColorHovered = colors.hovered;
+  const categoryColorBackground = colors.background;
```

Comment:
Nit: we could use directly `colors.XXX` fields instead of defining these intermediate variables.